doc/source/whatsnew/v0.25.0.rst

```diff
@@ -204,7 +204,7 @@ Interval
 Indexing
 ^^^^^^^^
 
--
+- Added an exception message when calling :meth:`DataFrame.iloc` with a list of non-numeric objects (:issue:`25753`).
```

Comment:
Could you reword to say `Improved exception message` instead of `Added an exception message`?  Technically, one existed it just wasn't that clear.

Reply:
Done!